apps/scoutgame/lib/builders/getLeaderboard.ts

```
@@ -23,9 +23,11 @@ export async function getLeaderboard({
   week?: string;
   season?: string;
 }): Promise<LeaderBoardRow[]> {
+  const isWeekValid = validateISOWeek(week);
+
   const weeklyTopBuilders = await prisma.userWeeklyStats.findMany({
     where: {
-      week,
+      week: isWeekValid ? week : getCurrentWeek(),
```

i think this method should throw an error instead of hide the invalid input

---

You're right. I have a default value in the page so this will not throw, but in the future we might reuse it so it is definitely better to add an error here.

I updated the func to throw if week is not valid!